simulator_service/src/simulator_service_node.cpp

```diff
@@ -0,0 +1,133 @@
+#include <ros/ros.h>
+
+#include <simulator_service/SimulateTrajectory.h>
+
+#include <actionlib/client/simple_action_client.h>
+#include <control_msgs/FollowJointTrajectoryAction.h>
+
+// Constants
+const static double ACTION_SERVER_WAIT_TIME = 5.0;
+const static double DEFAULT_SCALE_FACTOR = 0.2;
+
+// Utility Functions
+static void scaleDurations(trajectory_msgs::JointTrajectory& traj, double scale)
+{
+  for (std::size_t i = 0; i < traj.points.size(); ++i)
+  {
+    traj.points[i].time_from_start *= scale;
+  }
+}
+
+namespace simulator_service
+{
+  class SimulatorService
+  {
+  public:
+    SimulatorService(ros::NodeHandle& nh, const std::string& service_name, const std::string& action_name)
+      : ac_(action_name, true)
+      , service_name_(service_name)
+      , scale_factor_(DEFAULT_SCALE_FACTOR)
+    {
+      execute_trajectory_service_ = nh.advertiseService(service_name_, 
+        &SimulatorService::simulateTrajectoryCallback, 
+        this);
+
+      // Establish connection to robot action server
+      if (!ac_.waitForServer(ros::Duration(ACTION_SERVER_WAIT_TIME)))
+      {
+        std::ostringstream ss;
+        ss << "Could not connect to action: '" << action_name << "'";
+        ROS_ERROR_STREAM(ss.str());
+        throw std::runtime_error(ss.str());
```

Comment:
Throwing exceptions isn't endorsed by [google standards](https://google.github.io/styleguide/cppguide.html#Exceptions), however, since this is a self-contained node it may not matter much.